### PR TITLE
Fix getting stuck when returning to the main screen while the parameter editor is open

### LINKED ISSUE #9833: Parameter Editor "stuck" if you return to main view without closing it first
## Expected Behavior
I expect the Parameter Editor to automatically close if I click Back while it's open, or to not be allowed to click Back until I close it manually.

## Current Behavior
If I open the Parameter Editor to modify a parameter, and click Back without closing the Parameter Editor, I go back to the main view, but the Parameter Editor is still open, and it no longer responds to trying to close it by clicking the Cancel or Save buttons. It's also not possible to click the top left QGC icon button to go back to the parameter menu when in this state.

## Steps to Reproduce:
Please provide an unambiguous set of steps to reproduce the current behavior
1. Launch QGC and connect to a vehicle
2. Click the top left QGC icon and the open Vehicle Setup
3. Open the Parameters tab
4. Click a parameter to open the Parameter Editor pane on the right-hand side
5. The click Back in the top left corner
6. You return to the main view, but with the Parameter Editor still visible.

## System Information
When posting bug reports, include the following information
- Operating System: Ubuntu 18.04 (also seen on Android)
- QGC Version: Latest stable v4.1.3 and latest daily 048081c
- QGC build: Both daily and stable
- Flight Controller: Pixhawk 4 (in simulation and with hardware)
- Autopilot (with version): PX4 v1.11.0

## Detailed Description
N/A.

## Log Files and Screenshots
qgc

### PR DESCRIPTION
**Pull request by WTPENGUIN**
Fix #9833 

https://user-images.githubusercontent.com/28136137/131616675-400e376d-ee9a-4a76-9f1e-a02741398797.mp4

### CODE REVIEW

File: src/QmlControls/QGCViewDialogContainer.qml [hunk 1]
```diff
@@ -103,11 +103,25 @@ Drawer {
...
     }
 
+    Connections {
+        target: toolDrawer
+        onVisibleChanged: {
+            if(toolDrawer.visible === true) {
+                // Do nothing
+                return
+            }
+            else {
```
Comment (patrickelectric) [inline src/QmlControls/QGCViewDialogContainer.qml:113]: Else is unnecessary if inside the first if there is a return

**Review by patrickelectric — CHANGES_REQUESTED** [review]

**Comment by mortenfyhn** [comment]
Fantastic! :rocket:

**Comment by mortenfyhn** [comment]
I tested this myself and it certainly works! Would be great to get it merged.

**Comment by DonLakeFlyer** [comment]
@WTPENGUIN Is there a reason why this was closed?

**Comment by WTPENGUIN** [comment]
@DonLakeFlyer There was a request to change the code, so I thought I would close it and reopen it later.

**Comment by DonLakeFlyer** [comment]
Ah, ok thanks.

**Comment by DonLakeFlyer** [comment]
I re-opened since it fixes a critical bug. So might as well get it in. Not having the else is more a stylistic thing which to me can go either way. So I'll merge once it passes CI since I need this fix for stable.

**Comment by WTPENGUIN** [comment]
@DonLakeFlyer The CI test is a bit weird. A test failure occurs on the Mock Link side, and I have never modified the code there.

**Comment by DonLakeFlyer** [comment]
Hmm, there seems to be some flakiness in one of the unit tests. Looks completely unrelated to your change but re-running to be safe.

**Comment by WTPENGUIN** [comment]
@DonLakeFlyer Oh, there seems to be no problem.

**Comment by DonLakeFlyer** [comment]
Yup, thanks

**Comment by mortenfyhn** [comment]
Great! Thanks @WTPENGUIN

**Pull request merged**